### PR TITLE
add the useQueue Hook

### PR DESCRIPTION
**Pull request by MuriloucoLouco**

### CODE REVIEW

File: hooks/useQueue.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const [queue, setQueue] = useState(initialValue);
+
+  /**
+   * Function to add a value to the rear of the queue.
+   *
+   * @param {any} value Value to be added to the queue.
+   * @returns {undefined} This function returns nothing.
+   */
+  const enqueue = (value) => {
+    setQueue([value, ...queue]);
```
Comment (aromalanil) [inline hooks/useQueue.js:19]: Isn't rear supposed to be back of the queue like adding `enqueue(1)` to the queue `[2,6,7,9]` should give `[2,6,7,9,1]`

File: hooks/useQueue.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  /**
+   * Function to remove the value in the front of the queue.
+   *
+   * @returns {any} The value removed from the queue.
+   */
+  const dequeue = () => {
```
Comment (aromalanil) [inline hooks/useQueue.js:27]: Also the `dequeue()` should remove the element from the front. In this case, it is poping the last element.

**Comment by aromalanil** [comment]
@MuriloucoLouco  I have added some suggestions, please go through it. 
Also since this version adds a new feature, please change the package version to `1.7.0` in `package.json`

File: hooks/useQueue.js (hunk 1 above)
Comment (aromalanil) [inline hooks/useQueue.js:19]: This will be more correct for dequeue.
```suggestion
    setQueue([...queue, value]);
```

File: hooks/useQueue.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    if (queue.length > 0) {
+      const newQueue = [...queue];
+      const dequeuedValue = newQueue.pop();
```
Comment (aromalanil) [inline hooks/useQueue.js:30]: I think `shift` is the one to be used here instead of `pop`
```suggestion
      const dequeuedValue = newQueue.shift();
```

**Comment by MuriloucoLouco** [comment]
Oh, i'm sorry. I inverted the order of things.

**Comment by MuriloucoLouco** [comment]
Committed your revisions.

**Comment by aromalanil** [comment]
LGTM 👍🏻

**Comment by aromalanil** [comment]
@all-contributors please add @MuriloucoLouco  for code and doc

**Pull request merged**